src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1370,6 +1370,43 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
     return client.create_or_update(resource_group_name, name, instance, raw=no_wait)
 
 
+def _ensure_aks_service_principal(cli_ctx,
+                                  service_principal=None,
+                                  client_secret=None,
+                                  subscription_id=None,
+                                  dns_name_prefix=None,
+                                  location=None,
+                                  name=None):
+    file_name_aks = 'aksServicePrincipal.json'
+    # TODO: This really needs to be unit tested.
+    rbac_client = get_graph_rbac_management_client(cli_ctx)
+    if not service_principal:
+        # --service-principal not specified, try to load it from local disk
+        principal_obj = load_acs_service_principal(subscription_id, file_name=file_name_aks)
```

Comment:
If you like to differentiate terms between `acs` and `aks`, I suggest rename `load_acs_service_principal` to `load_aks_service_principal` since it is only being invoked inside `_ensure_aks_service_principal`, and the file name of `aksServicePrincipal.json` can be reasonably default'd inside the renamed method.

Reply:
actually, load_acs_service_principal is used by _ensure_aks_service_principal, and _ensure_service_principal.

Reply:
I commented under the context that we might be able to avoid cloning out the  `_ensure_aks_service_principal` altogether, but you have made the comment that the cloning is on purpose, so i am fine with it.